impeller/geometry/round_rect.h

```diff
@@ -0,0 +1,212 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
```

Comment:
I'm not sure what method we'll use for detection of SimpleRRects, but just a note that we should tolerate floating point errors on it.

Reply:
The rendering code already has its own NearlySimple method that I thought should be fine as that is a rendering issue. I originally had a method here that was similar, but then I decided to leave that to the caller since it wasn't clear what level of epsilon to use (and eventually we should be taking the matrix into account for any epsilon and that would complicate the methods even more).

Reply:
As mentioned down below, rather than have the rendering code do its own thing, the RoundRect method now applies a tolerance which can be supplied by the caller. We should supply a transform-based tolerance for rendering code and eventually we have to figure out what to do about all of these transform-dependent tolerances in the DL code that doesn't have access to the final transform.

Reply:
I like this approach!